### PR TITLE
[DataObjects] Migrations - fix error: All parts of a PRIMARY KEY must be NOT NULL

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves https://github.com/pimcore/pimcore/issues/11379

## Additional info  
follow up to #10760

### CODE REVIEW

**Comment by brusch** [comment]
@dvesh3 we need to fix that in the existing migration `Version20211103055110` as well, so that it can run through without issues 😊

**Comment by dvesh3** [comment]
> @dvesh3 we need to fix that in the existing migration `Version20211103055110` as well, so that it can run through without issues 😊

done  😉

File: bundles/CoreBundle/Migrations/Version20211103055110.php [hunk 1]
```diff
@@ -36,15 +36,11 @@ public function up(Schema $schema): void
 
             if ($schema->hasTable($objectDatastoreTableRelation)) {
                 $this->addSql(
-                    "ALTER TABLE $objectDatastoreTableRelation CHANGE COLUMN 
-                        `type` `type` VARCHAR(50)  NULL DEFAULT NULL ;"
+                    "UPDATE $objectDatastoreTableRelation SET `type` = NULL WHERE `type` = 'object'"
```
Comment (blankse) [inline bundles/CoreBundle/Migrations/Version20211103055110.php]:
```suggestion
                    "UPDATE $objectDatastoreTableRelation SET `type` = 'object' WHERE `type` = NULL"
```

Comment (blankse) [inline bundles/CoreBundle/Migrations/Version20211103055110.php:39]:
```suggestion
                    "UPDATE $objectDatastoreTableRelation SET `type` = 'object' WHERE `type` = NULL"
```

Comment (dvesh3) [reply bundles/CoreBundle/Migrations/Version20211103055110.php]: good catch! thanks 👍

Comment (dvesh3) [reply bundles/CoreBundle/Migrations/Version20211103055110.php]: I think the correct form is `WHERE type = ''` like using the previous default value.

File: bundles/CoreBundle/Migrations/Version20211103055110.php [hunk 2]
```diff
@@ -36,15 +36,11 @@ public function up(Schema $schema): void
 
             if ($schema->hasTable($objectDatastoreTableRelation)) {
                 $this->addSql(
-                    "ALTER TABLE $objectDatastoreTableRelation CHANGE COLUMN 
-                        `type` `type` VARCHAR(50)  NULL DEFAULT NULL ;"
+                    "UPDATE $objectDatastoreTableRelation SET `type` = 'object' WHERE `type` = NULL"
```
Comment (dvesh3) [inline bundles/CoreBundle/Migrations/Version20211103055110.php]:
```suggestion
                    "UPDATE $objectDatastoreTableRelation SET `type` = 'object' WHERE `type` = ''"
```

Comment (blankse) [inline bundles/CoreBundle/Migrations/Version20211103055110.php]:
```suggestion
                    "UPDATE $objectDatastoreTableRelation SET `type` = 'object' WHERE `type` = NULL OR `type` =''"
```

**Comment by brusch** [comment]
@abcpremium can you please confirm this patch does solve #11379  Thanks!

**Comment by abcpremium** [comment]
> @abcpremium can you please confirm this patch does solve #11379 Thanks!

It didn't unfortunately, I had to do these changes in vendor/pimcore/pimcore/bundles/CoreBundle/Migrations/Version20211103055110.php

There I did those changes and then it worked.

**Comment by brusch** [comment]
@abcpremium ok, so it did work, just just did the changes of this PR manually to `Version20211103055110` and then you ran it again? This would be ok, because this PR is still open and not part of a release yet 😉

**Comment by abcpremium** [comment]
> @abcpremium ok, so it did work, just just did the changes of this PR manually to `Version20211103055110` and then you ran it again? This would be ok, because this PR is still open and not part of a release yet 😉

Yes exactly, so I think this would solve the issue =)

**Pull request merged**